HumanRsourceManagementSystem/src/com/HRM/dao/DepartmentDaoImpl.java

```diff
@@ -0,0 +1,193 @@
+package com.HRM.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.HRM.dto.DepartmentDTO;
+import com.HRM.exception.DepartmentException;
+import com.HRM.utility.DBUtility;
+
+public class DepartmentDaoImpl implements DepartmentDao{
+
+	@Override
+	public String AddNewDepartments(DepartmentDTO dep) {
+		String msg = "adding process fail..";
+		
+	    try (Connection conn = DBUtility.ProvideConnection()){
+	    	
+	    	PreparedStatement ps = conn.prepareStatement("insert into department values(?,?,?)");
+	    	
+	    	ps.setInt(1, dep.getDepid());
+	    	ps.setString(2, dep.getDepnamne());
+	    	ps.setString(3, dep.getDepdesc());
+	    	
+	    	int  x = ps.executeUpdate();
+	    
+	    	if(x>0) {
+	    		msg = "department added successfully..";
+	    	}
+	    	
+	    	
+	    	
+	    }catch (SQLException e) {
+			e.printStackTrace();
+			msg = e.getMessage();
+			
+		}
+		
+		return msg;
+		
+	}
+
+	@Override
+	public List<DepartmentDTO> showDepartments() throws DepartmentException {
+
+        List<DepartmentDTO> dep = new ArrayList<>();
+		
+		try(Connection conn = DBUtility.ProvideConnection()){
+			
+			PreparedStatement ps =conn.prepareStatement("select * from department");

```

Comment:
there are many sql commands, cna you put these command in seperate file and use the one common function to use the same